addon/-private/data-view/radar/radar.js

```diff
@@ -259,19 +264,35 @@ export default class Radar {
       _componentPool.unshift(orderedComponents.pop());
     }
 
-    // If the underlying array has changed, the indexes could be the same but the content may have changed,
-    // so recycle the remaining components just in case. If content has not changed, this is a no-op.
-    for (let i = 0; i < orderedComponents.length; i++) {
-      const component = orderedComponents[i];
-      component.recycle(objectAt(items, component.index), component.index);
+    if (_wasReset) {
+      for (let i = 0; i < orderedComponents.length; i++) {
+        if (shouldRecycle === true) {
+          // If the underlying array has changed, the indexes could be the same but
+          // the content may have changed, so recycle the remaining components
+          const component = orderedComponents[i];
+          component.recycle(objectAt(items, component.index), component.index);
+        } else {
+          // If recycling is disabled we need to delete all components and clear the array
+          _componentPool.push(orderedComponents.shift());
```

Comment:
Will this successfully clear all components? Not sure about the effects of resizing an array while iterating over it.

```
a= [1,2,3,4,5]
> (5) [1, 2, 3, 4, 5]
for (let i = 0; i < a.length ; i++) {
  console.log(a.shift());
}
> 1
> 2
> 3
> undefined
```